public/app/features/playlist/PlaylistPage.tsx

```diff
@@ -1,34 +1,43 @@
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import { connect, MapStateToProps } from 'react-redux';
 import { NavModel } from '@grafana/data';
 import Page from 'app/core/components/Page/Page';
 import { StoreState } from 'app/types';
 import { GrafanaRouteComponentProps } from '../../core/navigation/types';
 import { getNavModel } from 'app/core/selectors/navModel';
-import { useAsync } from 'react-use';
+import { useAsync, useDebounce } from 'react-use';
 import { PlaylistDTO } from './types';
-import { Button, Card, ConfirmModal, LinkButton } from '@grafana/ui';
-import { contextSrv } from 'app/core/services/context_srv';
+import { ConfirmModal } from '@grafana/ui';
 import PageActionBar from 'app/core/components/PageActionBar/PageActionBar';
 import EmptyListCTA from '../../core/components/EmptyListCTA/EmptyListCTA';
 import { deletePlaylist, getAllPlaylist } from './api';
 import { StartModal } from './StartModal';
+import { PlaylistPageList } from './PlaylistPageList';
+import { EmptyQueryListBanner } from './EmptyQueryListBanner';
 
 interface ConnectedProps {
   navModel: NavModel;
 }
-
 export interface PlaylistPageProps extends ConnectedProps, GrafanaRouteComponentProps {}
 
 export const PlaylistPage: FC<PlaylistPageProps> = ({ navModel }) => {
   const [searchQuery, setSearchQuery] = useState('');
+  const [debouncedSearchQuery, setDebouncedSearchQuery] = useState(searchQuery);
+  const [hasFetched, setHasFetched] = useState(false);
   const [startPlaylist, setStartPlaylist] = useState<PlaylistDTO | undefined>();
   const [playlistToDelete, setPlaylistToDelete] = useState<PlaylistDTO | undefined>();
   const [forcePlaylistsFetch, setForcePlaylistsFetch] = useState(0);
 
   const { value: playlists, loading } = useAsync(async () => {
-    return getAllPlaylist(searchQuery) as Promise<PlaylistDTO[]>;
-  }, [forcePlaylistsFetch]);
+    return getAllPlaylist(searchQuery);
+  }, [forcePlaylistsFetch, debouncedSearchQuery]);
+  useEffect(() => {
```

Comment:
Nitpick: would place new line above this useEffect to make it easier to read the code.